multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -45,18 +142,470 @@ type ServiceExportReconciler struct {
+			err = r.handleServiceDeleteEvent(ctx, req, remoteCluster)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			r.installedSvcs.Delete(svcObj)
+		}
+
+		if epInstalled {
+			err = r.handleEndpointDeleteEvent(ctx, req, remoteCluster)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			r.installedEps.Delete(epObj)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// if corresponding service doesn't exist, update ServiceExport's status reason to not_found_service.
+	err = r.Client.Get(ctx, req.NamespacedName, svc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = r.updateSvcExportStatus(ctx, req, "not-found")
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		} else {
+			klog.ErrorS(err, "fail to get Service ", req.String())
+			return ctrl.Result{}, err
+		}
+	}
+
+	// skip if ServiceExport is trying to export MCS Service/Endpoints
+	if !svcInstalled || !epInstalled {
+		if _, ok := svc.Annotations[common.AntreaMCServiceAnnotation]; ok {
+			klog.InfoS("it's not allowed to export the multi-cluster controller auto-generated Service", "service", req.String())
+			err = r.updateSvcExportStatus(ctx, req, "imported-service")
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		}
+	}
+
+	// we also watch Service and Endpoints events via events mapping function
+	// need to check cache and compare with cache if there is any change for Service or Endpoints
+	var svcNoChange, epNoChange bool
+	svcExportNSName := common.NamespacedName(leaderNamespace, svcResExportName)
+	epExportNSName := common.NamespacedName(leaderNamespace, epResExportName)
+	if svcInstalled {
+		installedSvc := svcObj.(*svcInfo)
+		if apiequality.Semantic.DeepEqual(svc.Spec.Ports, installedSvc.ports) &&
+			apiequality.Semantic.DeepEqual(svc.Spec.ClusterIPs, installedSvc.clusterIPs) {
+			klog.InfoS("Service has been converted into ResourceExport and no change, skip it", "service",
+				req.String(), "resourceexport", svcExportNSName)
+			svcNoChange = true
+		}
+	}
+
+	ep := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      req.Name,
+			Namespace: req.Namespace,
+		},
+	}
+
+	err = r.Client.Get(ctx, req.NamespacedName, ep)
+	if err != nil {
+		klog.ErrorS(err, "fail to get Endpoints", "endpoints", req.String())
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if epInstalled {
+		installedEp := epObj.(*epInfo)
+		if apiequality.Semantic.DeepEqual(getEndPointsPorts(ep), installedEp.ports) &&
+			apiequality.Semantic.DeepEqual(getEndPointsAddress(ep), installedEp.addressIPs) {
+			klog.InfoS("Endpoints has been converted into ResourceExport and no change, skip it", "endpoints",
+				req.String(), "resourceexport", epExportNSName)
+			epNoChange = true
+		}
+	}
+
+	if epNoChange && svcNoChange {
+		return ctrl.Result{}, nil
+	}
+
+	re := mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: leaderNamespace,
+			Labels: map[string]string{
```

Comment:
K8s didn't provide annotation selector,  we rely on label selector to filter ResourceExport during ResorceImport computation, since it's created by our own process, I suppose it should be OK?